File: Session-3/DDD-Sample-codes/rtgs-return-management-dotnet10/Domain/Model/TransactionReference.cs
namespace Rtgs.ReturnManagement.Api.Domain.Model;

public sealed record TransactionReference
{
    public string Value { get; }

    public TransactionReference(string value)
    {
        if (string.IsNullOrWhiteSpace(value))
            throw new ArgumentException("Transaction reference is required.");
        Value = value;
    }
}

File: Session-3/DDD-Sample-codes/rtgs-return-management-springboot/src/main/java/com/bank/rtgs/returnmanagement/domain/model/TransactionReference.java
package com.bank.rtgs.returnmanagement.domain.model;

import java.util.Objects;

public record TransactionReference(String value) {
    public TransactionReference {
        Objects.requireNonNull(value, "Transaction reference is required.");
        if (value.isBlank()) throw new IllegalArgumentException("Transaction reference is required.");
    }
}
